From cf2434c417310703cfda61943c66f33fa23bde57 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Thu, 18 Aug 2016 09:48:29 +0300
Subject: [PATCH] Add aliases for open/read/write/close and setmode

---
 mingw-w64-crt/libarm32/msvcrt.def | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mingw-w64-crt/libarm32/msvcrt.def b/mingw-w64-crt/libarm32/msvcrt.def
index 2533d5a..4e3fe25 100644
--- a/mingw-w64-crt/libarm32/msvcrt.def
+++ b/mingw-w64-crt/libarm32/msvcrt.def
@@ -1278,3 +1278,8 @@ wprintf
 wprintf_s
 wscanf
 wscanf_s
+open = _open
+read = _read
+write = _write
+close = _close
+setmode = _setmode
-- 
2.7.4

