From 5592b70cca64935b0d090fe5030b3110ba17068c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Fri, 13 May 2016 15:17:56 +0200
Subject: [PATCH] yvals.h: Don't create a _Bool typedef if a macro already
 exists

---
 mingw-w64-headers/crt/yvals.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mingw-w64-headers/crt/yvals.h b/mingw-w64-headers/crt/yvals.h
index 70593c4..98b6921 100644
--- a/mingw-w64-headers/crt/yvals.h
+++ b/mingw-w64-headers/crt/yvals.h
@@ -163,10 +163,12 @@
 #define _Restrict __restrict__
 
 #ifdef __cplusplus
+#ifndef _Bool
 _STD_BEGIN
 typedef bool _Bool;
 _STD_END
 #endif
+#endif
 
 #define _LONGLONG /* __MINGW_EXTENSION */ __int64
 #define _ULONGLONG /* __MINGW_EXTENSION */ unsigned __int64
-- 
2.8.1

