From 2aa34e8e073d154d4e20286e5726f7670b1deee0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 31 Oct 2023 00:10:57 +0200
Subject: [PATCH 1/2] crt: Explicitly reference __C_specific_handler for Clang
 LTO

---
 mingw-w64-crt/crt/crtexe.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/mingw-w64-crt/crt/crtexe.c b/mingw-w64-crt/crt/crtexe.c
index 328cc305b..ca1d6d4bf 100644
--- a/mingw-w64-crt/crt/crtexe.c
+++ b/mingw-w64-crt/crt/crtexe.c
@@ -342,3 +342,14 @@ int __cdecl atexit (_PVFV func)
 }
 
 char __mingw_module_is_dll = 0;
+
+#if defined(__clang__) && defined(SEH_INLINE_ASM)
+// If building with LLVM LTO, the reference to __C_specific_handler in
+// .seh_handler in asm() above won't be visible to linking until actually
+// compiling all the LTO objects. If __C_specific_handler is provided as
+// an LTO object (as part of libucrtapp.a), it needs to be included directly.
+// Make a visible reference so that the linker knows that this function will
+// be needed.
+__attribute__((used))
+void *__personality_ptr = (void*)__C_specific_handler;
+#endif
-- 
2.43.0

